### PR TITLE
Added an option to hide the list of windows

### LINKED ISSUE #188: Hide tabs
Allow to hide tabs on tablayout and stack layout

### PR DESCRIPTION
**Pull request by nullgemm**
Hi,
This fixes #188 I hope it's clean enough :)

### CODE REVIEW

File: include/config_directives.h [hunk 1]
```diff
@@ -45,6 +45,7 @@ CFGFUN(for_window, const char *command);
 CFGFUN(gaps, const char *workspace, const char *type, const long value);
 CFGFUN(smart_borders, const char *enable);
 CFGFUN(smart_gaps, const char *enable);
+CFGFUN(hide_combined_titles, const char *enable);
```
Comment (Airblader) [inline include/config_directives.h]: I don't like this name too much. Can we change it to `force_tabbed_stacked_titlebar yes|no` with `yes` being the default?

File: include/configuration.h [hunk 2]
```diff
@@ -256,6 +256,9 @@ struct Config {
 
     /* Disable gaps if there is only one container on the workspace */
     bool smart_gaps;
+
+    /* Disable list of windows in stacked or tabbed mode */
```
Comment (Airblader) [inline include/configuration.h]: It's not so much a »list of windows«. It's the titlebar of the tabbed or stacked container.

File: src/render.c [hunk 3]
```diff
@@ -28,6 +28,8 @@ int render_deco_height(void) {
     int deco_height = config.font.height + 4;
     if (config.font.height & 0x01)
         ++deco_height;
+    if (config.hide_combined_titles)
```
Comment (Airblader) [inline src/render.c]: This is not the correct place to decide this. This option should only affect tabbed / stacked containers *and* only if their border is set to anything other than `normal` (i.e., either `pixel` or `none`). So I think the logic has to go into `con_render_tabbed` and similar.

Also, I think `con_border_style` has to be adapted as well. Once all of that is done please test this with various border settings and window layouts (nested tabbed containers, floating etc.) and different configs (enabled window borders etc.).

**Review by Airblader — CHANGES_REQUESTED** [review]
Thanks for your PR!

IMHO the implementation is currently not correct and there's also other minor things. On top of that, please also adapt the `README.md` to mention this feature in the appropriate spot.

Let me know when you pushed the changes so I can review again. Thanks!

**Comment by nullgemm** [comment]
I made the changes you requested and it seems to work fine in all cases. Should I pull to have the CI build passed ?

**Comment by Airblader** [comment]
> I made the changes you requested and it seems to work fine in all cases.

I don't see you ever setting the default value to `true`, so I assume that the option is enabled by default currently (which we don't want). Also, please add a section to the README explaining the feature. :-)

> Should I pull to have the CI build passed ?

Rebasing (not merging) the current gaps-next branch would be great, yes :-)

**Comment by nullgemm** [comment]
All done ! I noticed the smart gaps option does not have a default value in config.c btw, I'm curious: is there a specific reason to this ?

File: src/render.c [hunk 4]
```diff
@@ -452,11 +452,18 @@ static void render_con_stacked(Con *con, Con *child, render_params *p, int i) {
...
     child->deco_rect.width = child->rect.width;
-    child->deco_rect.height = p->deco_height;
 
-    if (p->children > 1 || (child->border_style != BS_PIXEL && child->border_style != BS_NONE)) {
+    if (config.force_tabbed_stacked_titlebar) {
+        child->deco_rect.y = p->y - con->rect.y + (i * p->deco_height);
+        child->deco_rect.height = p->deco_height;
+    } else {
+        child->deco_rect.y = p->y - con->rect.y;
+        child->deco_rect.height = (child->border_style == BS_PIXEL ? 1 : 0);
```
Comment (Airblader) [inline src/render.c:462]: Where does this `1` come from? Actually, this makes me think… shouldn't the descendants of stacked / tabbed containers also get the border on the top if `force_tabbed_stacked_titlebar` is disabled?

**Comment by Airblader** [comment]
> I noticed the smart gaps option does not have a default value in config.c btw, I'm curious: is there a specific reason to this?

The default of a boolean is false, respectively the default of an enum is 0. So if that's the intended default, it doesn't have to be set explicitly. Since we use a command here which has true as the default, we need to set it explicitly.

I'd like to change the README message, but I can do that after merging this. I do also have a couple more questions, see comment. Lastly, before we merge this we'll need to squash the commits, so you can do that already as well. Thanks!

**Comment by Airblader** [comment]
Hi, are you going to continue working on this?

**Comment by nullgemm** [comment]
Wow I forgot this I guess... I'll give it a try this afternoon and let you know if I can't get it to work properly. Thank you for your patience!

**Comment by nullgemm** [comment]
I fixed the top border not appearing and squashed+rebased, however I don't remember where that '1' in render.c came from...

**Comment by Airblader** [comment]
I think the `1` is wrong because it only accounts for `pixel 1` borders, not for `pixel 10` etc. And I also think to correctly show the top border for stacked / tabbed containers we'd probably need to make changes to things like `con_get_border_style_rect`, possibly other places as well. I haven't investigated this in detail, though.

**Comment by ifohancroft** [comment]
Are there any news on this? As in is it getting merged or did it get merged already? Does it work with the current version?

**Comment by Airblader** [comment]
You can see the status right here: it's still open and there are comments that haven't been addressed.

**Comment by ifohancroft** [comment]
You are right. Sorry, I missed that.

**Comment by Yohannfra** [comment]
Hi, are there any news on this ? Are you still working on it ?

**Pull request closed without merging**